### DISCUSSION TITLE
When to write integration tests

### DISCUSSION THREAD

**Discussion by m1yon** [quii/learn-go-with-tests#820] [Q&A]
Hey everyone!

I have a question on when to write integration tests. I've been writing a lambda that integrates two different services, and I feel like I might be writing too many integration tests.

I have contract integration tests for each service, which tests it against the live service and the fakes.

I also have an integration test for the service that uses both services (the output of service A gets fed into service B). This is almost like an acceptance test, but not as a black box because I'm still just calling the code directly. It has a lot of the same setup/assertions as the individual contract tests, so there's a decent amount of duplication here.

Now, I want to add an acceptance test that starts up a docker container, hits the live services, etc., but I feel like I might be writing too many integration tests.

Am I on the right track? Would anyone happen to have examples of an entire repository that implements a testing strategy aligned with learn go with tests?

**Comment by danon** [comment]
@m1yon I would recommend this source: https://www.youtube.com/watch?v=gnrBqLbj1_Q
